File: Algorithms/Easy/5/1022.SumofRootToLeafBinaryNumbers /main.go
package main

import "strconv"

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func goTree(node TreeNode, pref string) int {
	res := 0
	pref = pref + strconv.FormatInt(int64(node.Val), 10)
	if node.Left == nil && node.Right == nil {
		val, _ := strconv.ParseInt(pref, 10, 64)
		return res + int(val)
	}
	if node.Left != nil {
		res += goTree(*node.Left, pref)
	}
	if node.Right != nil {
		res += goTree(*node.Right, pref)
	}
	return res

}

func sumRootToLeaf(root *TreeNode) int {
	return goTree(*root, "")
}

File: Algorithms/Easy/5/1022.SumofRootToLeafBinaryNumbers /main.py
# Definition for a binary tree node.
from typing import *
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def sumRootToLeaf(self, root: Optional[TreeNode]) -> int:
        def go_tree(node: Optional[TreeNode], pref: str ='') -> None:
            res = 0
            pref = pref + str(node.val)
            if not node.left and not node.right:
                print(pref, int(pref, 2))
                return int(pref, 2)
            if node.left:
                res += go_tree(node.left, pref)
            if node.right:
                res += go_tree(node.right, pref)
            return res
        res = go_tree(root, '')
        return res